src/meson.build

```diff
@@ -0,0 +1,41 @@
+vkBasalt_src = [
+    'basalt.cpp',
+    'buffer.cpp',
+    'command_buffer.cpp',
+    'config.cpp',
+    'descriptor_set.cpp',
+    'effect_cas.cpp',
+    'effect.cpp',
+    'effect_deband.cpp',
+    'effect_fxaa.cpp',
+    'effect_lut.cpp',
+    'effect_reshade.cpp',
+    'effect_simple.cpp',
+    'effect_smaa.cpp',
+    'effect_transfer.cpp',
+    'fake_swapchain.cpp',
+    'format.cpp',
+    'framebuffer.cpp',
+    'graphics_pipeline.cpp',
+    'image.cpp',
+    'image_view.cpp',
+    'keyboard_input.cpp',
+    'logger.cpp',
+    'logical_swapchain.cpp',
+    'lut_cube.cpp',
+    'memory.cpp',
+    'renderpass.cpp',
+    'reshade_uniforms.cpp',
+    'sampler.cpp',
+    'shader.cpp',
+    'stb_image.cpp',
+    'stb_image_resize.cpp',
+    'util.cpp',
+]
+
+x11_dep = dependency('x11')
+
+shared_library(meson.project_name().to_lower(), 
```

Comment:
As far as I see it the lib isn't going to be installed. You would need to add `install: libdir`.
The libdir can be fetched with:
`libdir = get_option('libdir')`